chialuka/Banka
# #166171472- Disable client can create new staff and admin users

#### What does this PR do?
Prevents clients from opening new staff and admin user accounts

#### Description of Task to be completed?

- Provide authentication middleware that prevents clients from creating new staff and admin

- Ensure the route `/api/v1/admin` is protected from clients

- Ensure the route `/api/v1/staff` is protected from clients

#### How should this be manually tested?

- Using Postman or Insomnia, make a POST request to either `localhost:3000/api/v1/admin` or `localhost:3000/api/v1/staff`

- Include details of new staff to be created in the body of the request

- Send the request with a client's token in the authorization headers

- Notice that the response is a 403 and confirm from the Database records that a new user was not created

#### Any background context you want to provide?
The app allowed clients to create new staff and admin accounts. This is not acceptable because it would afford clients the opportunity to make themselves staff or admin on the app.
This commit fixes the bugs in the authentication middleware and routes that allowed this behavior.

#### What are the relevant pivotal tracker stories?
[[#166171056]](https://www.pivotaltracker.com/story/show/166171472)

#### Screenshots (if appropriate)

![Screenshot (100)](https://user-images.githubusercontent.com/42539315/58136620-9e6c2600-7c26-11e9-976b-33f24051a832.png)
